crates/sui-core/src/mysticeti_adapter.rs

```diff
@@ -101,18 +74,22 @@ impl SubmitToConsensus for LazyMysticetiClient {
         transaction: &ConsensusTransaction,
         _epoch_store: &Arc<AuthorityPerEpochStore>,
     ) -> SuiResult {
-        // The retrieved MysticetiClient can be from the past epoch. Submit would fail after
+        // TODO(mysticeti): confirm comment is still true
+        // The retrieved TransactionClient can be from the past epoch. Submit would fail after
         // Mysticeti shuts down, so there should be no correctness issue.
         let client = self.get().await;
+        let tx_bytes = bcs::to_bytes(&transaction).expect("Serialization should not fail.");
+        // TODO(mysticeti): convert consensus::ClientError -> SuiError properly
```

Comment:
What is the issue with the current approach?

Reply:
Forgot to remove this TODO as well